### DISCUSSION TITLE
How should I process a full prefetch before sending a multiple ack?

### DISCUSSION THREAD

**Discussion by thomasstep** [rabbitmq/amqp091-go#143] [Q&A]
The documentation says

> When multiple is true, this delivery and all prior unacknowledged deliveries on the same channel will be acknowledged. This is useful for batch processing of deliveries. 

Is this possible when using `Consume()` or do I need to use `Get()` to take advantage of multiple-ack? I'm trying to reduce total network round trips and I have successfully used multiple-ack in the past to do this (but with a different language/library).

**Comment by lukebakken** [comment]
> Is this possible when using Consume()

Yes, you can consume N messages the send the delivery tag of the Nth message with `multiple` set to true.

https://www.rabbitmq.com/confirms.html#consumer-acks-multiple-parameter

https://github.com/rabbitmq/amqp091-go/blob/main/confirms.go#L89-L100
